File: 35.search-insert-position.py
#
# @lc app=leetcode id=35 lang=python
#
# [35] Search Insert Position
#

# @lc code=start
class Solution(object):
    def searchInsert(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: int
        """
        try:
            return nums.index(target)
        except ValueError:
            if target < nums[0]:
                return 0
            elif target > nums[-1]:
                return len(nums)
            else:
                for i in range(1, len(nums)):
                    if nums[i] >= target and nums[i - 1] <= target:
                        return i
# @lc code=end

File: 35.search-insert-position.java
/*
 * @lc app=leetcode id=35 lang=java
 *
 * [35] Search Insert Position
 */

// @lc code=start
class Solution {
    public int searchInsert(int[] nums, int target) {
        try {
            for (int i = 0; i<nums.length; i++) {
                if (nums[i] == target) {
                    return i;
                }
            }
            throw new Exception();
        }
        catch (Exception e) {
            if (nums[0] > target) {
                return 0;
            }
            else if (nums[nums.length - 1]< target) {
                return nums.length;
            }
            else {
                for (int i = 1; i < nums.length; i++) {
                    if (nums[i]>=target && nums[i-1] <= target) {
                        return i;
                    }
                }
            }
        }
        return 0;
    }
}
// @lc code=end

